test/CMakeLists.txt

```diff
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.11)
 
 include(FetchContent)
-
+set(PISTACHE_LIBRARY ${FETCHCONTENT_BASE_DIR}/pistache-build/src/libpistache.so)
```

Comment:
When FetchContent runs, you can access symbols in the cmake files of the libraries that it brings in. Part of the Pistache project declares a library here: https://github.com/pistacheio/pistache/blob/master/src/CMakeLists.txt on line 18: `add_library(pistache OBJECT ${SOURCE_FILES})`. There's also a pistache_static target on line 58, which is what I ended up using for this phase: `add_library(pistache_static STATIC $<TARGET_OBJECTS:pistache>)`. These are both targets, and they can both be used as dependencies for a target that you're trying to build (by using `target_link_libraries`). That way you don't even have to worry about where in the directory tree the .so ends up being put, you can just refer to the target by name in cmake, and cmake will handle that for you.